hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestSCMPipelineManager.java

```diff
@@ -174,18 +174,17 @@ public void testPipelineReport() throws IOException {
     Assert
         .assertFalse(pipelineManager.getPipeline(pipeline.getId()).isOpen());
 
-    // get pipeline report from each dn in the pipeline
+    // pipeline is not healthy until all dns report
     PipelineReportHandler pipelineReportHandler =
         new PipelineReportHandler(scmSafeModeManager, pipelineManager, conf);
-    for (DatanodeDetails dn: pipeline.getNodes()) {
-      PipelineReportFromDatanode pipelineReportFromDatanode =
-          TestUtils.getPipelineReportFromDatanode(dn, pipeline.getId());
-      // pipeline is not healthy until all dns report
-      Assert.assertFalse(
-          pipelineManager.getPipeline(pipeline.getId()).isHealthy());
-      pipelineReportHandler
-          .onMessage(pipelineReportFromDatanode, new EventQueue());
-    }
+    List<DatanodeDetails> nodes = pipeline.getNodes();
```

Comment:
There is a similar negative case in Line 199-205. It does not cause failure but we can fix them similarly. The previous approach can also be fixed with slightly modification of TestUtils.getPipelineReportFromDatanode as demoed in the dup PR #195.